### PR TITLE
Adds "Stories" to Canon CMS

### LINKED ISSUE #691: implement story builder in CMS
This should be a new tab in the CMS (current interface would be called "Profilew"). Stories should be treated very similarly to Profiles, but they're a little less complicated.

Here's a checklist of features/structure:

* [x] add tab interface to admin panel ("Profiles" and "Stories" would be the initial 2 tabs)
* story
  * [x] title (one-line text input)
  * [x] slug (one-line text input) :trollface: 
  * [x] image (one-line text input)
  * [x] description(s)
  * [x] author(s)
    * [x] name (one-line text input)
    * [x] title (one-line text input)
    * [x] image (one-line text input)
    * [x] twitter (one-line text input)
    * [x] bio (quill editor)
  * [x] footnote(s) - would work similar to "dumb materialized generators", in that we should be able to add new footnotes that would have the following properties:
    * id - auto-incrementing from 1 (not user-defined), and would be a way for us place links within text (ie. maybe user types `Foot{{2}}` and it runs a formatter to turn that into an anchor link that goes to the footer of the page). This could be something we might want to bring to profile down the road, so let's come up with a solution that doesn't conflict with our existing language.
    * description (quill editor, only 1 per footnote)
  * [x] topic(s) - should have all the properties of a Profile topic:
    * [x] title
    * [x] slug
    * [x] type
    * [x] subtitle(s)
    * [x] stat(s)
    * [x] description(s)
    * [x] visualization(s)

### PR DESCRIPTION
**Pull request by jhmullen**
This PR closes #691.

This branch adds a second tab to the Canon CMS admin panel, "Stories."  It allows for the creation and management of stories, authors, footnotes, and "storytopics," a type of Topic that is relegated to a story only.

**Note**: After merging this, the CMS will no longer be located at `/admin/profilebuilder`, but rather at a new location:

`/admin/builder`

This type-agnostic naming was established to differentiate the ProfileBuilder from the StoryBuilder.

**Additional Note**:

This PR also adds a new endpoint for retrieving the stories, which may be accessed via `/api/story/:id`.

### CODE REVIEW

File: utils/stubs.js [hunk 1]
```diff
@@ -46,6 +46,33 @@ module.exports = {
     default: "",
     name: "new-selector"
   },
+  story_description: {
```
Comment (davelandry) [inline utils/stubs.js]: let's talk about `stubs`

Are these just default values for columns of a new row in the database? Can we just set the default values directly in the database schema and not have this file at all?

File: app/admin/components/TextCard.jsx [hunk 2]
```diff
@@ -102,6 +107,7 @@ class TextCard extends Component {
             onSave={this.save.bind(this)}
           />
         </Dialog>
+        { this.props.type === "story_footnote" && <div>Use Footnote Reference: <strong>{`Foot{{${this.props.ordering + 1}}}`}</strong></div>}
```
Comment (davelandry) [inline app/admin/components/TextCard.jsx]: You already have `type` from `this.props` on line 81, and you should also grab `ordering` as well.

File: api/mortarRoute.js [hunk 3]
```diff
@@ -247,4 +273,13 @@ module.exports = function(app) {
       });
   });
 
+  // Endpoint for getting a story
+  app.get("/api/story/:id", (req, res) => {
```
Comment (davelandry) [inline api/mortarRoute.js]: This directly conflicts with the route defined in `api/storyRoute.js`. Please do the following:
1. compare this route with the older route, and make sure it is returning similar structure (off the cuff, I know you'll have to parse out the `date` from the slug (slugs look like this: `08-11-2016_rust-belt`)
2. hook up an additional route that matches the current `/api/story` route in `api/storyRoute.js` (for the "featured" flag, now that we have explicit ordering in the CMS just always make the top story (order `0`) the featured one)

File: utils/stubs.js (hunk 1 above)
Comment (jhmullen) [reply utils/stubs.js]: Yes, you are correct. The original purpose of this file was for making *tree elements*, which needed additional metadata (repeated, visual properties like `hasCaret`).  But I followed this down to database elements, and DEFAULT values in the schema would suffice.  I will make these changes.

**Comment by jhmullen** [comment]
Updates to this branch:
- slugs have been added to stories, and `/api/canonstory/:id` may be searched using either a traditional story slug or the direct numeric id
- both `/api/canonstory/:id` and `/api/canonstory` now include a `date` field that is parsed from the beginning of the slug, identical to the way the current route works
- default values have been added to all db models (so adding them at the psql level is not necessary)

Please review!

**Pull request merged**